File: app-mvvm/src/main/java/com/xuie/imaginary/base/BaseBindingViewHolder.java
package xuk.imaginary.base;

import android.view.View;

import androidx.databinding.ViewDataBinding;

import com.chad.library.adapter.base.BaseViewHolder;

/**
 * Date: 2017/5/11 14:42.
 * Email: tyshengsx@gmail.com
 *
 * @author tysheng
 */
public class BaseBindingViewHolder<Binding extends ViewDataBinding> extends BaseViewHolder {
    private Binding mBinding;

    public BaseBindingViewHolder(View view) {
        super(view);
    }

    public Binding getBinding() {
        return mBinding;
    }

    public void setBinding(Binding binding) {
        mBinding = binding;
    }
}

File: app-coroutines/src/main/kotlin/xuk/imaginary/base/BaseBindingViewHolder.kt
package xuk.imaginary.base

import android.view.View

import androidx.databinding.ViewDataBinding

import com.chad.library.adapter.base.BaseViewHolder

/**
 * Date: 2017/5/11 14:42.
 * Email: tyshengsx@gmail.com
 *
 * @author tysheng
 */
class BaseBindingViewHolder<Binding : ViewDataBinding>(view: View) : BaseViewHolder(view) {
  var binding: Binding? = null
}
